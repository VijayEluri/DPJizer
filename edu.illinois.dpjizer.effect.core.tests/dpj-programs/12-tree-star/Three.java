/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package three;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C<region X> {
	region L;
	region Q;
	C<X:L> left in X:L;
	int a in X;

	C() {}

	void m1() {
		left.m2();
	}

	void m2() {
		a = 3;
		new C<Q>().m1();
	}
}