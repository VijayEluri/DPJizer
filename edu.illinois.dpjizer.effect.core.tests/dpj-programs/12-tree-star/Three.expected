three.C.C() invokes java.lang.Object.Object()
three.C.m1() reads X : three.C.L invokes three.C.m2() where { this -> X : three.C.L : *, X -> X : three.C.L }
three.C.m2() writes X invokes three.C.m1() where { this -> Root : three.C.Q : *, X -> Root : three.C.Q }
